lib/adapters/definition-adapter.ts

```diff
@@ -68,11 +68,15 @@ export default class DefinitionAdapter {
    * @param locationResult Either a single {Location} object or an {Array} of {Locations}.
    * @returns An {Array} of {Location}s or {null} if the locationResult was null.
    */
-  public static normalizeLocations(locationResult: Location | Location[]): Location[] | null {
+  public static normalizeLocations(
+    locationResult: Location | Location[] | LocationLink[] | null
+  ): Location[] | LocationLink[] | null {
     if (locationResult == null) {
       return null
     }
-    return (Array.isArray(locationResult) ? locationResult : [locationResult]).filter((d) => d.range.start != null)
+    return (Array.isArray(locationResult) ? (locationResult as any[]) : [locationResult]).filter(
+      (d: Location | LocationLink) => ("range" in d ? d.range : d.targetRange).start != null
```

Comment:
Because the array is not a mixed array, this check should only be performed for the first element of the array. There is no need to check it for every element.